### PR TITLE
Feature/eng 00006 docker utils

### PR DESCRIPTION
**Pull request by bayoumi17m**
# Overview
Added docker utility functions!

## API Updates
- N/A
 
### New Features *(required)*
- Added internal API to build docker containers

### Deprecations *(required)*
- N/A

### Enhancements *(required)*
- N/A

## Checklist
- [X] Interrogate has > 70% coverage
- [ ] All tests have passed (new tests are made)
- [ ] Total test coverage > 75%
- [ ] New code test coverage > 80%
- [ ] Pass all linting checks

### CODE REVIEW

File: src/coalescenceml/utils/docker_utils.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        logger.warning(
+            f"Unable to find dockerignore file at path '{dockerignore_path}'"
+        )
+        return []
+
+    exclude_patterns = []
+    for line in file_content.split("\n"):
+        line = line.strip()
+        if line and not line.startswith("#"):
+            exclude_pattens.append(line)
```
Comment (fort745) [inline src/coalescenceml/utils/docker_utils.py:46]: typo

File: src/coalescenceml/utils/docker_utils.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+    return exclude patterns
```
Comment (fort745) [inline src/coalescenceml/utils/docker_utils.py:48]: typo

File: src/coalescenceml/utils/docker_utils.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+                else:
+                    pass
+            except json.JSONDecodeError as error:
+                logger.warning(
+                    f"Failed to decode json for line '{line}' with "
+                    f"error: {error}"
+                )
+
+
+def push_docker_image(image_name: str) -> None:
```
Comment (fort745) [inline src/coalescenceml/utils/docker_utils.py:177]: not currently implemented yet

**Comment by JSun14** [comment]
looks good to me aside from the typos caught

**Review by bayoumi17m — COMMENTED** [review]
Manually tested each function and it worked for me! Let me know of any issues with docs etc or if we can merge this tonight!

**Pull request merged**